freqtrade/strategy/hyper.py

```diff
@@ -0,0 +1,213 @@
+        """
+        Initialize hyperopt-optimizable parameter.
+        :param space: A parameter category. Can be 'buy' or 'sell'. This parameter is optional if
+         parameter field
+         name is prefixed with 'buy_' or 'sell_'.
+        :param optimize: Include parameter in hyperopt optimizations.
+        :param load: Load parameter value from {space}_params.
+        :param kwargs: Extra parameters to skopt.space.(Integer|Real|Categorical).
+        """
+        if 'name' in kwargs:
+            raise OperationalException(
+                'Name is determined by parameter field name and can not be specified manually.')
+        self.category = space
+        self._space_params = kwargs
+        self.value = default
+        self.opt_range = opt_range
+        self.optimize = optimize
+        self.load = load
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.value})'
+
+    @abstractmethod
+    def get_space(self, name: str) -> Union['Integer', 'Real', 'Categorical']:
+        """
+        Get-space - will be used by Hyperopt to get the hyperopt Space
+        """
+
+
+class IntParameter(BaseParameter):
+    default: int
+    value: int
+    opt_range: Sequence[int]
+
+    def __init__(self, low: Union[int, Sequence[int]], high: Optional[int] = None, *, default: int,
+                 space: Optional[str] = None, optimize: bool = True, load: bool = True, **kwargs):
+        """
+        Initialize hyperopt-optimizable parameter.
+        :param low: Lower end (inclusive) of optimization space or [low, high].
+        :param high: Upper end (inclusive) of optimization space.
+                     Must be none of entire range is passed first parameter.
+        :param default: A default value.
+        :param space: A parameter category. Can be 'buy' or 'sell'. This parameter is optional if
+                      parameter fieldname is prefixed with 'buy_' or 'sell_'.
+        :param optimize: Include parameter in hyperopt optimizations.
+        :param load: Load parameter value from {space}_params.
+        :param kwargs: Extra parameters to skopt.space.Integer.
+        """
+        if high is not None and isinstance(low, Sequence):
+            raise OperationalException('IntParameter space invalid.')
+        if high is None or isinstance(low, Sequence):
+            if not isinstance(low, Sequence) or len(low) != 2:
+                raise OperationalException('IntParameter space must be [low, high]')
+            opt_range = low
+        else:
+            opt_range = [low, high]
+        super().__init__(opt_range=opt_range, default=default, space=space, optimize=optimize,
+                         load=load, **kwargs)
+
+    def get_space(self, name: str) -> 'Integer':
+        """
+        Create skopt optimization space.
+        :param name: A name of parameter field.
+        """
+        return Integer(*self.opt_range, name=name, **self._space_params)
+
+
+class FloatParameter(BaseParameter):
```

Comment:
i keep asking myself if we should add another parameter type with limited decimals.

The FloatParameter is currently using the `Real` space - which has a (in theory) unlimited precision (i guess it'll be 16 decimals ...).

This is a problem in it's own right and can lead to a "vanishing gradient" problem - where the changes hyperopt performs on a parameter are too small to be visible.  

e.g. if you compare something to 30.1111 or 30.11125123123123 - will not make your results better or worse - only overfitted.

So my idea would be:
Rename the "existing" space to `RealParameter` (aligned with the `Real` space)

Add a parameter to `FloatParameter` called decimals (defaults to 3?) which will:
`pow` the values by decimals for the space and uses the `Integer` space
`unpow` the values in a `value` property

so for example the following space 
`FloatParameter([0, 1], decimals=2)` would only have the following possible values (100 in total):
`[0.01, 0.02, 0.03, ...0.98, 0.99]`

Reply:
Ohh that is a good idea. I'll do it 👍🏻